Jenkinsfile

```diff
@@ -628,3 +629,71 @@ def isChanged(patterns){
     || isGitRegionMatch(patterns: ["^libbeat/.*"], comparator: 'regexp')
   )
 }
+
+def loadConfigEnvVars(){
+  env.BUILD_AUDITBEAT = isChanged(["^auditbeat/.*"])
+  env.BUILD_AUDITBEAT_XPACK = isChanged([
+    "^auditbeat/.*",
+    "^x-pack/auditbeat/.*",
+    "^x-pack/libbeat/.*",
+  ])
+  env.BUILD_DOCKERLOGBEAT_XPACK = isChanged([
+    "^x-pack/dockerlogbeat/.*",
+    "^x-pack/libbeat/.*",
+  ])
+  env.BUILD_DOCS = isGitRegionMatch(
+    patterns: ["^docs/.*"],
+    comparator: 'regexp'
+  ) || params.runAllStages
+  env.BUILD_FILEBEAT = isChanged(["^filebeat/.*"])
+  env.BUILD_FILEBEAT_XPACK = isChanged([
+    "^filebeat/.*",
+    "^x-pack/filebeat/.*",
+    "^x-pack/libbeat/.*",
+  ])
+  env.BUILD_FUNCTIONBEAT_XPACK = isChanged([
+    "^x-pack/functionbeat/.*",
+    "^x-pack/libbeat/.*",
+  ])
+  env.BUILD_GENERATOR = isChanged(["^generator/.*"])
+  env.BUILD_HEARTBEAT = isChanged(["^heartbeat/.*"])
+  env.BUILD_HEARTBEAT_XPACK = isChanged([
+    "^heartbeat/.*",
+    "^x-pack/heartbeat/.*",
+    "^x-pack/libbeat/.*",
+  ])
+  env.BUILD_JOURNALBEAT = isChanged(["^journalbeat/.*"])
+  env.BUILD_JOURNALBEAT_XPACK = isChanged([
+    "^journalbeat/.*",
+    "^x-pack/journalbeat/.*",
+    "^x-pack/libbeat/.*",
+  ])
+  env.BUILD_KUBERNETES = isChanged(["^deploy/kubernetes/*"])
+  env.BUILD_LIBBEAT = isGitRegionMatch(
+    patterns: ["^libbeat/.*"],
+    comparator: 'regexp'
+  ) || params.runAllStages
+  env.BUILD_LIBBEAT_XPACK = isChanged([
+    "^libbeat/.*",
```

Comment:
Is this one needed? I think it's part of `isChanged`.

Reply:
I forget to refactor this, I will remove the isChanged method, to avoid hide some matching pattern, today I will finish it.

Reply:
finally, I make it the other way around, we use always the `isChanged` that would make the path match and check the run all stages parameter, this is less verbose.